app/scripts/app.js

```diff
@@ -1,170 +1,182 @@
-      player.animation.y = y;
-    }
-  }
+  // set the canvas to the full page size
+  c.width = screenWidth;
+  c.height = screenHeight;
 
-  function keyDown(evt) {
-     keysdown[evt.which] = true;
-
-    if (keysdown[38] && keysdown["37"]) {
-      console.log('up & left');
-      player.pos.x -= 4;
-      player.pos.y -= 4;
-      player.animation.y = 0;
-    } else if (keysdown[38] && keysdown[39]) {
-      console.log('up & right');
-      player.pos.x += 4;
-      player.pos.y -= 4;
-      player.animation.y = 130;
-    } else if (keysdown[40] && keysdown[37]) {
-      console.log('down and left');
-      player.pos.x -= 4;
-      player.pos.y += 4;
-      player.animation.y = 327;
-    } else if (keysdown[40] && keysdown[39]) {
-      console.log('down and right');
-      player.pos.x += 4;
-      player.pos.y += 4;
-      player.animation.y = 454;
-    } else if (keysdown[38] === true) {
-      console.log('up');
-      player.animation.y = 64;
-      player.pos.y -= 4;
-      player.animation.y = 65;
-    }
-    else if (keysdown[40] === true) {
-      console.log('down');
-      player.pos.y += 4;
-      player.animation.y = 391;
-    }
-    else if (keysdown[37] === true) {
-      console.log('left');
-      player.pos.x -= 4;
-      player.animation.y = 260;
-    }
-    else if (keysdown[39] === true) {
-      console.log('right');
-      player.pos.x += 4;
-      player.animation.y = 194;
-    }
-  }
+  // keeps track of which keys are pressed
+  var keyState = [];
 
-  function keyUp(evt) {
-    keysdown[evt.which] = false;
-  }
+  window.addEventListener('keydown', function (e) {
+    keyState[e.keyCode || e.which] = true;
+  }, true);
 
-  function Keyboarder() {
-    this.isDown = function(keyCode) {
-      return keyState[keyCode] === true;
-    };
+  window.addEventListener('keyup', function (e) {
+    keyState[e.keyCode || e.which] = false;
+  }, true);
 
-    this.KEYS = {
-      UP: 38,
-      DOWN: 40,
-      LEFT: 37,
-      RIGHT: 39,
-      SPACE: 32
+  function Player(totalHealth, currentHealth) {
+    this.health = {
+      totalHealth: totalHealth,
+      currentHealth: currentHealth,
+      percent: (currentHealth / totalHealth) * 100,
     };
-  }
-
-  function Player() {
-    this.keyboarder = new Keyboarder();
     this.animation = {
       x: 0,
-      y: 0
+      y: 0,
     };
     this.frame = {
-      currentFrame: 0
+      current_frame: 0,
+      frames: 9,
+      width: (500 / 9), // 55.555555555...6
+      height: (519 / 9), // 57.6666666...7
     };
     this.pos = {
-      x: 0,
-      Y: 0
+      x: screenWidth / 2 - ((500 / 9) / 2),
+      y: screenHeight / 2  - ((519 / 9) / 2),
     };
     this.speed = 1;
-    this.avatar = new Image();
+    this.avatar;
   };
 
-  Player.prototype.update = function() {
-    if (this.keyboarder.isDown(this.keyboarder.KEYS.LEFT)) {
-      this.pos.x -= 2;
+  Player.prototype.update = function (timeBetweenFrames) {
+    if (keyState[38] && keyState[37]) {
```

Comment:
You mean keyState[KEY_UP]? KEY_UP is just a variable assigned to a number. The variable names are just used to add more readability when accessing the array of booleans keyState[], with indices corresponding to their respective key codes. Also, the key code 37 is left, not right